cmd/srcd/cmd/parse.go

```diff
@@ -216,12 +220,25 @@ func parseLang(ctx context.Context, client api.EngineClient, path string, b []by
 	return res.Lang, nil
 }
 
-func isSupportedLanguage(supportedDrivers []*api.ListDriversResponse_DriverInfo, desired string) bool {
+type languageSupportInfo struct {
+	supported   []string
+	requested   string
+	isSupported bool
+}
+
+func getLanguagesSupportInfo(supportedDrivers []*api.ListDriversResponse_DriverInfo, desired string) *languageSupportInfo {
```

Comment:
For some reason this feels odd. The struct return type is there to provide the info you need to build the error message, only in case of error.
Instead of that, we could build the nice error message inside this function. Something like:
```go
func checkSupportedLanguage(supportedDrivers []*api.ListDriversResponse_DriverInfo, desired string) error
```

Reply:
done, I already squashed it as it is a very small thing.